core-rust/state-manager/src/receipt.rs

```diff
@@ -36,6 +37,18 @@ impl CommittedTransactionIdentifiers {
     }
 }
 
+#[derive(Debug, Clone, ScryptoCategorize, ScryptoEncode, ScryptoDecode)]
+pub struct ApplicationEvent {
+    pub type_id: EventTypeIdentifier,
+    pub data: Vec<u8>,
+}
+
+impl ApplicationEvent {
+    pub fn new(type_id: EventTypeIdentifier, data: Vec<u8>) -> Self {
+        Self { type_id, data }
+    }
+}
```

Comment:
Ditto - as per my comment on the other PR - I wonder if it's worth us explicitly spelling out how the receipt event hash is constructed in a method here?

Reply:
I've now added a few more rustdocs 👍